src/App.vue

```diff
@@ -239,67 +374,74 @@ export default {
       a.download = fileName + '.' + (ext || 'json')
       a.target = '_self'
       a.click()
-    }
-  },
-  data () {
-    return {
-      socket: null,
-      pages: [
-        { icon: 'widgets', title: 'Control Panel', path: '/' },
-        { icon: 'settings', title: 'Settings', path: '/settings' },
-        { icon: 'movie_filter', title: 'Scenes', path: '/scenes' },
-        { icon: 'bug_report', title: 'Debug', path: '/debug' },
-        { icon: 'folder', title: 'Store', path: '/store' },
-        { icon: 'share', title: 'Network graph', path: '/mesh' }
-      ],
-      settings: new Settings(localStorage),
-      status: '',
-      statusColor: '',
-      drawer: false,
-      mini: false,
-      topbar: [],
-      title: '',
-      snackbar: false,
-      snackbarText: '',
-      dark: undefined,
-      baseURI: ConfigApis.getBasePath()
-    }
-  },
-  watch: {
-    $route: function (value) {
-      this.title = value.name || ''
     },
-    dark (v) {
-      this.settings.store('dark', this.dark)
+    async startSocket () {
+      if (this.socket || !this.$route.meta || !this.$route.meta.requiresAuth) {
+        return
+      }
 
-      this.$vuetify.theme.dark = v
-      this.changeThemeColor()
-    }
-  },
-  beforeMount () {
-    this.title = this.$route.name || ''
+      if (!this.user || !this.user.token) {
+        await this.logout()
+      }
 
-    const self = this
+      this.socket = io('/', {
+        path: ConfigApis.getSocketPath(),
+        query: { token: this.user.token }
+      })
 
-    this.socket = io('/', {
-      path: ConfigApis.getSocketPath()
-    })
+      this.socket.on('connect', () => {
+        this.updateStatus('Connected', 'green')
+      })
 
-    this.socket.on('connect', () => {
-      self.updateStatus('Connected', 'green')
-    })
+      this.socket.on('disconnect', () => {
+        this.updateStatus('Disconnected', 'red')
+      })
 
-    this.socket.on('disconnect', () => {
-      self.updateStatus('Disconnected', 'red')
-    })
+      this.socket.on('error', () => {
+        console.log('Socket error')
```

Comment:
mix of `console.log` and `self.updateStatus`?

Reply:
nope I simply ignore the socket error as it's an intermediate state before the reconnect